doc/migration.md

```diff
@@ -0,0 +1,93 @@
+# Hypersync SDK Migration Guide
+
+## Version 2.X to Version 3.0
+
+Version 3.0 of the Hypersync SDK includes a number of new capabilities that
+require small changes to Hypersync apps that use version 2.1 or earlier.
+
+### Package Versions
+
+Hypersync SDK functionality is now distributed via three Node packages:
+
+| Package                      | Version | Description                 |
+| ---------------------------- | ------- | --------------------------- |
+| @hyperproof/hyperysnc-sdk    | 3.0.0   | Core Hypersync SDK          |
+| @hyperproof/hypersync-models | 5.0.0   | Supporting Hypersync models |
+| @hyperproof/integration-sdk  | 1.0.2   | Common integration elements |
+
+Most Hypersync apps will require all three packages to function properly.
+
+The `dependencies` section of your app's `package.json` file should be updated as follows:
+
+```
+  "dependencies": {
+    "@hyperproof/integration-sdk": "^1.0.2",
+    "@hyperproof/hypersync-models": "^5.0.0",
+    "@hyperproof/hypersync-sdk": "^3.0.0",
+    ...
+  }
+```
+
+### Imports
+
+As noted above and in the README, a number of classes, functions, types, interfaces and enums
+been moved to `@hyperproof/integration-sdk`. Hypersync SDK developers should take note of the
+following language elements which must now be imported from `@hyperproof/integration-sdk`.
+
+If your app makes use of any of these elements, your `import` statements will need to be updated.
+
+| Element               | Type  |
+| --------------------- | ----- |
+| CustomAuthCredentials | Type  |
+| Logger                | Class |
+
+### Updated Execution Environment
+
+Hypersync apps in Hyperproof EU execute in an updated execution environment. A couple of
+changes need to be made to existing Hypersync apps to allow them to run in Hyperproof EU.
+
+The primary difference between the Hyperproof execution environment and the Hyperproof EU
+execution environment is the way that Hypersync apps are started. Hyperproof EU requires
+an `up` script in `package.json` which creates an initializes an HTTP server in the app.
```

Comment:
the updated execution environment is also planned in the future for the regular hyperproof.app, may be worth mentioning that as well so people can prep for it.

Reply:
Think our plan needs to become more firm before I say anything about that.